### ISSUE TITLE
Error in training

### ISSUE THREAD

**Issue by floew2** [floew2/aircraft-detection-deeplearning#2]
python -m src.models.trainer data.yaml --epochs 50 --plot throws an erorr:

python -m src.models.trainer data.yaml --epochs 50 --plot
/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py:127: RuntimeWarning: 'src.models.trainer' found in sys.modules after import of package 'src.models', but prior to execution of 'src.models.trainer'; this may result in unpredictable behaviour
  warn(RuntimeWarning(msg))
2025-08-15 07:44:51,276 - INFO - Configuration loaded from params.yaml
Starting training with data: data.yaml
Configuration: params.yaml
Overrides: {'epochs': 50}
2025-08-15 07:44:51,277 - INFO - Starting YOLOv8 training...
2025-08-15 07:44:51,277 - INFO - Loading model: yolov8s.pt
2025-08-15 07:44:53,036 - INFO - Loaded pretrained model: yolov8s.pt
2025-08-15 07:44:53,036 - INFO - Training configuration: {'epochs': 50, 'batch': 16, 'imgsz': 512, 'patience': 50, 'device': 'auto', 'workers': 8, 'optimizer': 'auto', 'lr0': 0.01, 'momentum': 0.937, 'weight_decay': 0.0005, 'warmup_epochs': 3, 'warmup_momentum': 0.8, 'warmup_bias_lr': 0.1, 'box': 7.5, 'cls': 0.5, 'dfl': 1.5, 'save': True, 'save_period': -1, 'cache': False, 'rect': False, 'cos_lr': False, 'close_mosaic': 10, 'resume': False, 'amp': True, 'fraction': 1.0, 'profile': False, 'freeze': None, 'multi_scale': False, 'overlap_mask': True, 'mask_ratio': 4, 'dropout': 0.0, 'val': True}
2025-08-15 07:44:53,036 - INFO - Project: runs/detect, Name: train_20250815_074453
Ultralytics 8.3.179 🚀 Python-3.8.10 torch-2.2.2 
Error: Training failed: Invalid CUDA 'device=auto' requested. Use 'device=cpu' or pass valid CUDA device(s) if available, i.e. 'device=0' or 'device=0,1,2,3' for Multi-GPU.

torch.cuda.is_available(): False
torch.cuda.device_count(): 0
os.environ['CUDA_VISIBLE_DEVICES']: None
See https://pytorch.org/get-started/locally/ for up-to-date torch install instructions if no CUDA devices are seen by torch.
